tests/test_operator.py

```diff
@@ -150,17 +150,13 @@ class TestArguments(object):
     def setup_class(cls):
         clear_cache()
 
-    def test_override(self):
-        """Test that the call-time overriding of Operator arguments works"""
+    def test_override_symbol(self):
+        """Test call-time symbols overrides with other symbols"""
         i, j, k, l = dimify('i j k l')
-        a = symbol(name='a', dimensions=(i, j, k, l), value=2.,
-                   mode='indexed').base.function
-        a1 = symbol(name='a', dimensions=(i, j, k, l), value=3.,
-                    mode='indexed').base.function
-        a2 = symbol(name='a', dimensions=(i, j, k, l), value=4.,
-                    mode='indexed').base.function
-        eqn = Eq(a, a+3)
-        op = Operator(eqn)
+        a = symbol(name='a', dimensions=(i, j, k, l), value=2.)
+        a1 = symbol(name='a1', dimensions=(i, j, k, l), value=3.)
+        a2 = symbol(name='a2', dimensions=(i, j, k, l), value=4.)
+        op = Operator(Eq(a, a + 3))
```

Comment:
I think this test should go back to what it was until we decide to explicitly prohibit multiple symbols of the same name. The change you are making here has already been added as a new test in #255 . Just merge that here.